### PR TITLE
support wider range of clang versions

### PR DESCRIPTION
**Pull request by skimo-openhub**

### CODE REVIEW

File: interface/generator.cc [hunk 1]
```diff
@@ -391,6 +391,25 @@ bool generator::first_arg_is_isl_ctx(FunctionDecl *fd)
 	return is_isl_ctx(param->getOriginalType());
 }
 
+namespace {
+
+struct ClangAPI {
```
Comment (ftynse) [inline interface/generator.cc]: This can be a (non-anonymous) namespace instead

**Comment by skimo-openhub** [comment]
On Fri, Jul 13, 2018 at 06:31:29AM -0700, ftynse wrote:
> ftynse approved this pull request.
> > @@ -391,6 +391,25 @@ bool generator::first_arg_is_isl_ctx(FunctionDecl *fd)
>  	return is_isl_ctx(param->getOriginalType());
>  }
>  
> +namespace {
> +
> +struct ClangAPI {
> 
> This can be a (non-anonymous) namespace instead

Hmm... I was just trying to mimic the ClangAPI in extract_interface.cc.
I'm not sure why a struct was used there.  I didn't write that part and
I didn't question it when it was introduced.

Maybe a namespace is cleaner...

skimo

**Comment by skimo-openhub** [comment]
On Fri, Jul 13, 2018 at 03:40:18PM +0200, Sven Verdoolaege wrote:
> On Fri, Jul 13, 2018 at 06:31:29AM -0700, ftynse wrote:
> > ftynse approved this pull request.
> > > @@ -391,6 +391,25 @@ bool generator::first_arg_is_isl_ctx(FunctionDecl *fd)
> >  	return is_isl_ctx(param->getOriginalType());
> >  }
> >  
> > +namespace {
> > +
> > +struct ClangAPI {
> > 
> > This can be a (non-anonymous) namespace instead
> 
> Hmm... I was just trying to mimic the ClangAPI in extract_interface.cc.
> I'm not sure why a struct was used there.  I didn't write that part and
> I didn't question it when it was introduced.
> 
> Maybe a namespace is cleaner...

Actually, trying it out now, I remember I tried it out before.
Using struct avoid the generation of an unused function warning.

skimo

**Comment by ftynse** [comment]
Ok, although it's inconsistent for a compiler to warn about unused functions but not about unused member functions when the class is wrapped in an anonymous namespace.

**Pull request merged**